src/utils.c

```diff
@@ -173,16 +181,16 @@ void pm(int M, int N, float *A)
 
 void find_replace(const char* str, char* orig, char* rep, char* output)
 {
-    char* buffer = (char*)calloc(8192, sizeof(char));
+    char* buffer = strdup(str);
```

Comment:
Does `p` points to `buffer` which points to `str`?

What will be done in this line `*p = '\0';` ?

Reply:
strdup allocates and copies  a string,so p points to a heap string.

Reply:
@cyyever Yes, you are right! ) https://stackoverflow.com/a/252802/1558037